misc/inmanta-workon-register.sh

```diff
@@ -152,61 +151,50 @@ function __restore_old_config {
         unset PIP_CONFIG_FILE
     fi
 
-    echo "Done restoring old config..." >&2
     return 0
 }
 
-function __get_pip_config_setting {
-    declare result
-    if [ "$1" == "extra_index_url" ] ; then
-        # make sure extra index url are formatted correctly (space-separated) e.g. "idx0 idx1 idx2"
-        result=$(
-            "$INMANTA_WORKON_PYTHON" -c "from inmanta.module import Project; project=Project('.', autostd=False); pip_cfg=project.metadata.pip;print(' '.join(pip_cfg.$1));" #2> /dev/null TODO add error suppression back in ?
-        )
-        echo "$result"
-        return 0
-    fi
-    if [ "$1" == "index_url" ] ; then
-        result=$(
-            "$INMANTA_WORKON_PYTHON" -c "from inmanta.module import Project; project=Project('.', autostd=False); pip_cfg=project.metadata.pip;print(pip_cfg.$1) if pip_cfg.$1 else ...;" #2> /dev/null
-        )
-        echo "$result"
-        return 0
-    fi
-    if  [ "$1" == "pre" ] || [ "$1" == "use_system_config" ] ; then
-        result=$(
-            "$INMANTA_WORKON_PYTHON" -c "from inmanta.module import Project; project=Project('.', autostd=False); pip_cfg=project.metadata.pip;print(pip_cfg.$1);" #2> /dev/null
-        )
-        echo "$result"
-        return 0
-    fi
-
-    echo "ERROR: invalid pip config setting $1" >&2
 
-    return 1
-}
+function __get_pip_config {
+    python_script=$(cat << END
+from inmanta.module import Project, ProjectConfigurationWarning
+import warnings
+warnings.filterwarnings("error", category=ProjectConfigurationWarning)
+try:
+    project=Project('.', autostd=False)
+except ProjectConfigurationWarning:
+    exit(1)
+pip_cfg=project.metadata.pip.model_dump()
+for k in ['pre','index_url','use_system_config']:
+    print(pip_cfg[k]) if pip_cfg[k] is not None else print('')
+print(' '.join(pip_cfg['extra_index_url']))
+END
+)
+
+    result=$("$INMANTA_WORKON_PYTHON" -c "${python_script}") >&2
```

Comment:
I don't think this `>&2` does anything because we capture the stdout in `result` anyway.